plugins/container_logs.yaml

```diff
@@ -138,16 +138,43 @@ template: |
             from: attributes.log
             to: body
 
+          # Initial severity is derived from the stream field, which can be
+          # "stderr" or "stdout"
+          - type: severity_parser
+            parse_from: attributes.stream
+            mapping:
+              info: stdout
+              error: stderr
+
           # Parse body as json if the body appears to be json
           {{ if .body_json_parsing }}
           - type: json_parser
             if: body matches "^{.*}\\s*$"
             parse_from: body
             parse_to: body
+
+          # After parsing the log body as json, parse severity
+          # or level if present, with severity taking precedence.
+          - type: router
+            default: file_name_parser
+            routes:
+              - expr: 'body.severity != nil'
+                output: body_severity_parser
+              - expr: 'body.level != nil'
+                output: body_level_parser
```

Comment:
Are these mutally exclusive?

Reply:
I think so. Im not sure I have observed logs which have both fields. Google looks for `severity`. I added `level` because it is common.

There is probably a lot of room for improvement here, but picking these two fields was low hanging fruit.